File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        cur_ele = -101
        idx = -1
        for i in nums:
            if cur_ele != i:
                idx+=1
                nums[idx] = i
                cur_ele = i

            
        return idx+1
        

File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    void solve(int idx,vector<int>&nums,int t,vector<vector<int>>&res,vector<int>&path)
    {
        if(t == 0)
        {
            res.push_back(path);
            return;
        }
        if(idx>=nums.size())
        {
            return;
        }
        //pick
        if(nums[idx] <= t)
        {
            path.push_back(nums[idx]);
            solve(idx,nums,t - nums[idx],res,path);
            path.pop_back();
        }
        solve(idx+1,nums,t,res,path);
    }
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<vector<int>>res;
        vector<int>path;
        solve(0,nums,target,res,path);
        return res; 
    }
};
